src/test/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTaskTest.java

```diff
@@ -812,15 +812,26 @@ public void testLoggedTime() throws IOException {
         final long startTime = System.currentTimeMillis();
         antTask.execute();
         final long endTime = System.currentTimeMillis();
+        final long testingTime = endTime - startTime;
         final List<MessageLevelPair> loggedMessages = antTask.getLoggedMessages();
-        final long loggedTimeInExecute =
-            getNumberFromLine(loggedMessages.get(loggedMessages.size() - 1).getMsg());
-
-        assertWithMessage("Logged time inside method cannot be more than "
-                              + "logged time outside the method")
-            .that(loggedTimeInExecute)
-            .isAtMost(endTime - startTime);
 
+        assertLoggedTime(loggedMessages, testingTime, "Total execution");
+        assertLoggedTime(loggedMessages, testingTime, "To locate the files");
+        assertLoggedTime(loggedMessages, testingTime, "To process the files");
+    }
+
+    private static void assertLoggedTime(List<MessageLevelPair> loggedMessages,
+                                         long testingTime, String expectedMessage) {
+        loggedMessages.stream()
+            .filter(msg -> msg.getMsg().startsWith(expectedMessage))
+            .findFirst()
+            .ifPresentOrElse(msg -> {
```

Comment:
Do we need conditions in this assert method? Can we just have one assertion to make test fail somehow?

Reply:
In the test cases that we have, optional will always contain a value, but I think it's better to use this than just `.get()` and one `assert` condition. This will also provide future safety (in case logged messages change in the future, why the test is failing will be clear) and this method can also be used by other tests in the future.

Reply:
Please separate conditions to eliminate `ifPresentOrElse`  and hard to read code:

```java
    private static void assertLoggedTime(List<MessageLevelPair> loggedMessages,
                                         long testingTime, String expectedMsg) {

        final Optional<MessageLevelPair> optionalMessageLevelPair = loggedMessages.stream()
                .filter(msg -> msg.getMsg().startsWith(expectedMsg))
                .findFirst();

        assertWithMessage("Message should be present.")
                .that(optionalMessageLevelPair.isPresent())
                .isTrue();

        final long actualTime = getNumberFromLine(optionalMessageLevelPair.get().getMsg());

        assertWithMessage("Logged time in '" + expectedMsg + "' "
                + "must be less than the testing time")
                .that(actualTime)
                .isAtMost(testingTime);
    }


```

Reply:
Done.